include/maidsafe/crux/detail/multiplexer.hpp

```diff
@@ -200,26 +208,44 @@ inline void multiplexer::remove(socket_base *socket)
     }
 }
 
-template <typename SocketType,
+template <typename AcceptorType,
+          typename SocketType,
           typename AcceptHandler>
-void multiplexer::async_accept(SocketType& socket,
+void multiplexer::async_accept(AcceptorType* acceptor,
+                               SocketType& socket,
                                AcceptHandler&& handler)
 {
-    std::unique_ptr<accept_input_type> operation(new accept_input_type(&socket,
+    std::unique_ptr<accept_input_type> operation(new accept_input_type(acceptor,
+                                                                       &socket,
                                                                        std::move(handler)));
-    acceptor_queue.emplace(std::move(operation));
+    acceptor_queue.emplace_back(std::move(operation));
+
+    start_receive();
+}
 
-    // We need to start receiving on this multiplexer through the
-    // socket's idempotent_start_receive method (and not through our
-    // start_receive one) to make the socket aware it has started
-    // receiving (as one socket may start only one receive at a time).
-    socket.idempotent_start_receive();
+inline
+void multiplexer::disable_accept_requests_from(acceptor* accept) {
+    auto i = acceptor_queue.begin();
+
+    while (i != acceptor_queue.end()) {
```

Comment:
Can there be multiple occurrences of the same acceptor within the queue?

If not, then you could use `find()` instead.